### ISSUE TITLE
Use happypack with ExtractTextPlugin

### ISSUE THREAD

**Issue by XVincentX** [amireh/happypack#12]
Would it be possible to provide an example about how to use HappyPack with `ExtractTextPlugin`?

**Comment by amireh** [comment]
So the plugin itself should work if you just feed it the loaders you're already feeding to happypack (I'm only assuming) but I know for sure that less/style/css loaders do NOT work right now because they're using the pitch methods.

I've started a branch to address this, and I did get part of it to work but I'm not sure if I want to go through with it it because it's making things really complicated.

How many style files do you have?

**Comment by XVincentX** [comment]
A lot (477 to be precise)

**Comment by amireh** [comment]
Ok yeah that might be worth it then. We use SASS but without webpack.

Any chance your project is open that I may test against?

**Comment by XVincentX** [comment]
Unfortunately it is not open and I am not allowed to open in :-1:

**Comment by amireh** [comment]
Ok, no problem.

I'll update this ticket when there's something ready to be tested, but it's probably gonna be a while.

**Comment by XVincentX** [comment]
Absolutely, thanks a lot! Looking forward

**Comment by texttechne** [comment]
:+1: for supporting ExtractTextPlugin and Style loaders

Sass is also the main bottle neck in our project builds...

**Comment by amireh** [comment]
I've some free time tomorrow, will work on this and see if it can be done without a lot of piggybacking against webpack's compilation.

**Comment by amireh** [comment]
So, now that style/css/sass loaders are working (in #15), I've taken a look at the source of extract-text-webpack-plugin and unfortunately, **I don't think it will ever be supported in this architecture**.

The loader itself generates a child compiler and hooks into the main compilation, while the plugin reaches out to the loader(s). The loaders modify their own contexts at run-time and actually modify the entirety of the chunk and its assets, then they also use `this.exec` to generate some more source... I don't even know what it's supposed to do, and even if I thought I did, I've less of an idea as to why it has to do all of that to do what it says it does.

TL;DR - I don't have the time to bother with that plugin. Perhaps consider not using webpack for your stylesheets; at my work we don't and it's just fine (~273 sass files) - we use plain node-sass behind a grunt task.

**Comment by amireh** [comment]
Sorry guys, gonna close this one.

**Comment by amireh** [comment]
Can you guys check my comment in #14 - https://github.com/amireh/happypack/issues/14#issuecomment-208254692 about using ExtractText and see if that trick makes it work for you?

**Comment by XVincentX** [comment]
Absolutely! Should I just take last version from npm? I cannot find any particular branch for now.

**Comment by amireh** [comment]
Yes! Anything > 2.0 should be ok!

**Comment by XVincentX** [comment]
I did not go through all my application, but it looks like it it is working at the moment.
